src/controllers/handlers/utils.ts

```diff
@@ -28,6 +26,58 @@ export async function validate<T extends string>(
   return {
     identity,
     realmName,
-    sceneId
+    sceneId,
+    parcel,
+    hostname
+  }
+}
+
+export function formatUrl(url: string): string {
+  if (!url) return '/'
+  return url.endsWith('/') ? url : `${url}/`
+}
+
+export function isValidAddress(address: string): boolean {
+  if (typeof address !== 'string' || !address) return false
+  return /^0x[a-fA-F0-9]{40}$/i.test(address)
+}
+
+export function validateFilters(filters: { admin?: string }): {
+  valid: boolean
+  error?: string
+  value: { admin?: string }
+} {
+  if (filters.admin !== undefined && typeof filters.admin !== 'string') {
+    return {
+      valid: false,
+      error: 'admin must be a string',
+      value: {}
+    }
+  }
+
+  return {
+    valid: true,
+    value: {
+      admin: filters.admin ? filters.admin.toLowerCase() : undefined
+    }
+  }
+}
+
+export async function fetchDenyList(): Promise<Set<string>> {
+  try {
+    const response = await fetch('https://config.decentraland.org/denylist.json')
```

Comment:
There is no config for this? will we always use the same URL?

Reply:
About this fetchDenyList I just moved the function, but let me add that into envs